test/Libraries/Microsoft.Extensions.Http.Resilience.Tests/Resilience/HttpClientBuilderExtensionsTests.Resilience.cs

```diff
@@ -286,6 +287,78 @@ public void AddResilienceHandler_AuthorityByCustomSelector_NotValidated()
         Assert.NotNull(factory.CreateClient("my-client"));
     }
 
+    [Fact]
+    public void RemoveAllResilienceHandlers_ArgumentValidation()
+    {
+        var services = new ServiceCollection();
+        IHttpClientBuilder? builder = null;
+        Assert.Throws<ArgumentNullException>(() => builder!.AddResilienceHandler("pipeline-name", _ => { }));
+        Assert.Throws<ArgumentNullException>(() => builder!.AddResilienceHandler("pipeline-name", (_, _) => { }));
+    }
+
+    [Fact]
+    public void RemoveAllResilienceHandlers_EnsureHandlersRemoved()
+    {
+        var services = new ServiceCollection();
+
+        IHttpClientBuilder? builder = services.AddHttpClient("custom");
+
+        builder.AddStandardResilienceHandler();
+
+        builder.ConfigureAdditionalHttpMessageHandlers((handlers, _) =>
+        {
+            Assert.Single(handlers);
+        });
+
+        builder.RemoveAllResilienceHandlers();
+
+        builder.ConfigureAdditionalHttpMessageHandlers((handlers, _) =>
+        {
+            Assert.Empty(handlers);
+        });
+
+        using ServiceProvider serviceProvider = services.BuildServiceProvider();
+        services.BuildServiceProvider().GetRequiredService<IHttpClientFactory>().CreateClient("custom");
```

Comment:
```suggestion
        serviceProvider.BuildServiceProvider().GetRequiredService<IHttpClientFactory>().CreateClient("custom");
```

We already have the `serviceProvider` so we can use it.

Reply:
Please, fix similar issue in the tests below too.

Reply:
Done